### PR TITLE
Feature/client response

### LINKED ISSUE #29: Upload full schema using the CLI
Currently, you can only upload schemas per table, it would be nice to be able to upload the full schema.

### LINKED ISSUE #30: Make queries using the CLI
I would like to be able to make queries using the CLI.

### LINKED ISSUE #40: Quick-add-table API
Add the `Quick-add-table` API to SDK, similar to the `quick-add-table` command in CLI

### PR DESCRIPTION
**Pull request by tringm**
### Architectureal changes:
- Added a `JsonFormat` class interface that uses jsonschema to validate json objects. This is used in the Schema objects
- Added the `client_request` and `client_response` module which contains different request and response class.
- **AitoClient.request** now either take an instance of AitoRequest or create a Request object from  `method`, `endpoint`, `query` arguments. 
- The AitoClient now also returns an enriched Response object instead of JSON as before.
- Refactor helper methods of the **AitoClient** to the **aito.api** module

### New features:
- Both **AitoClient** and the **request** method now have the **raise_for_status** option to control either raise or return an Error object.
- Added methods in the **AitoClient**  to send a query to the API Endpoint
- Added **quick_add_table** API
- Added **create-database** command, added commands to send a query to the API Endpoint
- Use requirements range instead of fixed versions. Remove the **--encoding** flag in the **convert** and **infer-table-schema** command to support xlrd 1.2
- Use safety to security check dependencies
- *Beta features*: 
  - Added 2 new API functions: **quick_predict** and **quick_predict_and_evaluate** to generate an example Predict query
  - Added the **quick-predict** command to generate an example predict query and evaluate its performance

### CODE REVIEW

File: aito/api.py [hunk 1]
```diff
@@ -0,0 +1,763 @@
...
+    :param client: the AitoClient object
+    :type client: AitoClient
+    :param database_schema: Aito database schema
+    :type database_schema: Dict
+    :return: the database schema
+    :rtype: Dict
+    """
+    if not isinstance(database_schema, AitoDatabaseSchema):
+        database_schema = AitoDatabaseSchema.from_deserialized_object(database_schema)
+    client.request(method='PUT', endpoint='', query=database_schema.to_json_serializable())
```
Comment (laaksonm) [inline aito/api.py]: Shouldn't the endpoint here be `/api/v1/schema`?

File: aito/cli/sub_commands/database_sub_command.py [hunk 2]
```diff
@@ -352,9 +352,80 @@ def parse_and_execute(self, parsed_args: Dict):
...
+    @abstractmethod
+    def endpoint(self) -> str:
+        pass
+
+    def __init__(self):
+        super().__init__(self.endpoint, f'send a query to the {self.endpoint.upper()} API')
+
+    def build_parser(self, parser):
+        parser.add_aito_default_credentials_arguments()
+        parser.add_argument('query', type=str, help='the query to be send')
```
Comment (laaksonm) [inline aito/cli/sub_commands/database_sub_command.py]:
```suggestion
        parser.add_argument('query', type=str, help='the query to be sent')
```

File: docs/source/changelog.rst [hunk 3]
```diff
@@ -1,6 +1,58 @@
 Changelog
 =========
 
+0.4.0
+-----
+
+This version contains changes on how you make requests with the :py:class:`~aito.client.AitoClient`.
+In addition to specifying the request method, endpoint as before, you can now use different
```
Comment (laaksonm) [inline docs/source/changelog.rst]:
```suggestion
In addition to specifying the request method and endpoint as before, you can now use different
```

File: docs/source/changelog.rst [hunk 4]
```diff
@@ -1,6 +1,58 @@
...
+:py:mod:`Request objects <aito.client_request>` or use the endpoint methods of the client,
+i.e: :py:func:`aito.client.AitoClient.predict`. The client now returns the enriched
+:py:mod:`Response objects <aito.client_response>` instead of JSON as before.
+
+Helper methods of the AitoClient, e.g: create_table, are moved to the :py:mod:`aito.api` module.
+
+You can now execute requests from the CLI
```
Comment (laaksonm) [inline docs/source/changelog.rst]:
```suggestion
You can now execute requests from the CLI.
```

File: docs/source/changelog.rst [hunk 5]
```diff
@@ -1,6 +1,58 @@
...
+
+
+SDK
+^^^
+
+- Schema objects validation in the :py:mod:`aito.schema` module is improved
```
Comment (laaksonm) [inline docs/source/changelog.rst]:
```suggestion
- Schema objects validation in the :py:mod:`aito.schema` module is improved.
```

File: docs/source/changelog.rst [hunk 6]
```diff
@@ -1,6 +1,58 @@
...
+- Added the :py:mod:`aito.client_request` module which contains the request objects for :py:class:`~aito.client.AitoClient`.
+- Added the :py:mod:`aito.client_response` module which contains the response objects returned by the :py:class:`~aito.client.AitoClient`.
+
+AitoClient
+""""""""""
+
+  - The **AitoClient.request** method no longer takes positional argument. You now have to specify either `method`, `endpoint` or `request_obj`.
```
Comment (laaksonm) [inline docs/source/changelog.rst]:
```suggestion
  - The **AitoClient.request** method no longer takes positional arguments. You now have to specify either `request_obj` or `method` and `endpoint`.
```

File: docs/source/changelog.rst [hunk 7]
```diff
@@ -1,6 +1,58 @@
...
+  - Both **AitoClient** and **AitoClient.request** now has a `raise_for_status` argument which controls whether the client should raise or return an **aito_client.RequestError** object when an error occurred during sending a request.
```
Comment (laaksonm) [inline docs/source/changelog.rst]:
```suggestion
  - Both **AitoClient** and **AitoClient.request** now have a `raise_for_status` argument which controls whether the client should raise or return an **aito_client.RequestError** object when an error occurs during sending a request.
```

**Review by laaksonm — COMMENTED** [review]
Some slight fixes and one question. :nerd_face:

File: aito/api.py (hunk 1 above)
Comment (tringm) [reply aito/api.py]: Good spot

**Review by laaksonm — APPROVED** [review]
Amazing :star_struck:

**Pull request merged**